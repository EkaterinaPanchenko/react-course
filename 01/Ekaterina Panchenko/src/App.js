import { Header } from "./componets/header/Header";
import { Main } from "./componets/main/Main";
import { Footer } from "./componets/footer/Footer";
import { config } from "./componets/config";

export const App = () => (
  <>
    <Header menu={config.menu} contact={config.contact} />
    <Main buttonContent={config.buttonContent} mainTitle={config.mainTitle} />
    <Footer footerContact={config.footerContact} />
  </>
);
