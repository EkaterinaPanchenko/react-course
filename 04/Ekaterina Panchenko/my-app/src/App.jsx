import React from "react";

import Header from "./components/Header";
import Img from "./components/Img";

const dog = [
  { name: "akita", id: 1 },
  { name: "chow", id: 2 },
  { name: "boxer", id: 3 },
  { name: "husky", id: 4 },
  { name: "mix", id: 5 },
];

class App extends React.Component {
  state = {
    activeImg: "",
  };

  handleFetch = (name) => {
    fetch(`https://dog.ceo/api/breed/${name}/images/random`)
      .then((res) => res.json())
      .then(({ message }) => {
        alert("Изображение загрузилось!");
        this.setState({
          activeImg: message,
        });
      });
  };

  handleChange = (name) => {
    this.handleFetch(name);
  };

  componentDidMount() {
    this.handleFetch("australian/shepherd");
  }

  render() {
    return (
      <>
        <Header handleChange={this.handleChange} dog={dog} />
        <Img src={this.state.activeImg} />
      </>
    );
  }
}

export default App;
