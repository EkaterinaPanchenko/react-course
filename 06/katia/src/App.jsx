import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Map from "./components/Map.jsx";

class App extends React.Component {
  render() {
    return (
      <Router>
        <Switch>
          <Route path="/" exact component={Map} />
          <Route path="/:left/:top" component={Map} />
        </Switch>
      </Router>
    );
  }
}

export default App;
