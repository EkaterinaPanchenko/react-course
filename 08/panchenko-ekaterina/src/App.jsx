import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import NameCar from "./componets/NameCar.jsx";
import DescriptionCar from "./componets/DescriptionCar.jsx";
import styled from "styled-components";
import DataContextProvider from "./componets/DataContextProvider";

const App = () => {
  return (
    <Router>
      <Container>
        <DataContextProvider>
          <NameCar />
          <DescriptionCar />
        </DataContextProvider>
      </Container>
    </Router>
  );
};

const Container = styled.div`
  display: flex;
`;

export default App;
