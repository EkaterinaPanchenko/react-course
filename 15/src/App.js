import { Route, Switch } from 'react-router-dom';

import { 
  Home, 
  Navigation,
  Civilizations,
  Civilization,
  Structures,
  Structure, 
  Technologies,
  Technology, 
  Units,
  Unit
} from './components';

import { 
  GlobalStyle, 
  Wrapper, 
  SwitchWrapper 
} from './globalStyle';
  
function App() {

  return (
  <Wrapper>
      <GlobalStyle />
      <Navigation />
    <SwitchWrapper>
      <Switch>
        <Route exact path='/' component={Home}/>
        <Route exact path='/civilizations' component={Civilizations}/>
        <Route exact path='/civilizations/:civilizationType' component={Civilization}/>
        <Route exact path='/structures' component={Structures}/>
        <Route exact path='/structures/:name-:structureId' component={Structure}/>
        <Route exact path='/technologies' component={Technologies}/>
        <Route exact path='/technologies/:name-:technologyId' component={Technology}/>
        <Route exact path='/units' component={Units}/>
        <Route exact path='/units/:name-:unitId' component={Unit}/>
      </Switch>
    </SwitchWrapper>
  </Wrapper>
  )
};
    

export default App;
