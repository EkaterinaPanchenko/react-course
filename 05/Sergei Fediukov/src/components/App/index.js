import './App.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'

import Routes from '../Routes';
import FooterLogOut from '../FooterLogOut';
import MainNavigation from '../MainNavigation'
import LoaderPrimary from '../Loader/LoaderPrimary';

export function App({ email }) {
  return (
    <div className="App">
      {Boolean(email) &&
        <>
          <MainNavigation />
          <LoaderPrimary />
          <hr />
          <FooterLogOut />
        </>
      }
      <Routes />
    </div>
  );
}
const mapStateToProps = state => {
  return {
    email: state.firebase.auth.email
  };
};

App.propTypes = {
  name: PropTypes.string
}
App.defaultProps = {
  email: undefined
}
export default connect(mapStateToProps)(App);
