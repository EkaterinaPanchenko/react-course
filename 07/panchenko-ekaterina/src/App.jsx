import React from "react";
import { CheckIn, Login, AccountDeleting } from "./components/Forms.jsx";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import { Form } from "./components/Form";
import styled from "styled-components";

const App = () => {
  return (
    <Router>
      <div className="App">
        <StyledLink to="/check-in">CheckIn</StyledLink>
        <StyledLink to="/login">Login</StyledLink>
        <StyledLink to="/account-deleting">AccountDeleting</StyledLink>

        <Form
          render={() => (
            <Switch>
              <Route exact path="/" component={CheckIn} />
              <Route path="/check-in" component={CheckIn} />
              <Route path="/login" component={Login} />
              <Route path="/account-deleting" component={AccountDeleting} />
            </Switch>
          )}
        />
      </div>
    </Router>
  );
};

const StyledLink = styled(Link)`
  margin-left: 40px;
`;

export default App;
