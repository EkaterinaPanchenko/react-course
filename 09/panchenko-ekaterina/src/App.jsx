import React from "react";
import { CheckIn, Login, AccountDeleting } from "./components/Forms.jsx";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import { Form } from "./components/Form";
import styled from "styled-components";

const App = () => (
  <Router>
    <div className="App">
      <StyledLink to="/check-in">CheckIn</StyledLink>
      <StyledLink to="/login">Login</StyledLink>
      <StyledLink to="/account-deleting">AccountDeleting</StyledLink>

      <Form
        render={(props) => (
          <Switch>
            <Route exact path="/">
              <CheckIn {...props} />
            </Route>
            <Route path="/check-in">
              <CheckIn {...props} />
            </Route>
            <Route path="/login">
              <Login {...props} />
            </Route>
            <Route path="/account-deleting">
              <AccountDeleting {...props} />
            </Route>
          </Switch>
        )}
      />
    </div>
  </Router>
);

const StyledLink = styled(Link)`
  margin-left: 40px;
`;

export default App;
