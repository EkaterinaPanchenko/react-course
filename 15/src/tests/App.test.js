import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import App from '../App';

configure({ adapter: new Adapter() });

describe('<Home/> unit test', () => {
    const wrapper = shallow(<App/>);

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});