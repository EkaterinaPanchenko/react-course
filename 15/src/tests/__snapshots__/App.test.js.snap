// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> unit test should match snapshot 1`] = `
<styled.div>
  <Memo(l) />
  <Navigation />
  <styled.div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/civilizations"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/civilizations/:civilizationType"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/structures"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/structures/:name-:structureId"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/technologies"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/technologies/:name-:technologyId"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/units"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/units/:name-:unitId"
      />
    </Switch>
  </styled.div>
</styled.div>
`;
